creator/ingest_runs/models/ingest_run.py

```diff
@@ -0,0 +1,56 @@
+from django.contrib.auth import get_user_model
+from django.db import models
+
+from creator.ingest_runs.common.model import IngestProcess
+from creator.files.models import Version
+
+DELIMITER = "-"
+NAME_PREFIX = "INGEST_RUN"
+INGEST_QUEUE_NAME = "ingest"
+
+User = get_user_model()
+
+
+class IngestRun(IngestProcess):
+    """
+    Request to ingest file(s) into a target data service
+    """
+
+    class Meta(IngestProcess.Meta):
+        permissions = [
+            ("list_all_ingestrun", "Show all ingest_runs"),
+            ("cancel_ingestrun", "Cancel an ingest_run"),
+        ]
+    name = models.TextField(
+        blank=True,
+        null=True,
+        help_text=(
+            "The name of the ingest run. Autopopulated on save with the "
+            "concatenation of the IngestRun's file version IDs"
+        ),
+    )
+    versions = models.ManyToManyField(
+        Version,
+        related_name="ingest_runs",
+        help_text="List of files to ingest in the ingest run",
+    )
+
+    @property
+    def study(self):
+        if self.versions.count() > 0:
+            return self.versions.first().root_file.study
+
+    def compute_name(self):
```

Comment:
Why not make this a property? If there is a name, return it, if not, compute and save it then return it.

Reply:
I think I wanted the `name` field back when I was developing the ingest run model and needed it for debugging. I didn't make it a property bc I wanted it to be a field that got persisted in the db so I could filter by name in the queries. I think we could prob just get rid of it now?

Reply:
Ah right. You could have the database field called `_name` and then use the `name` property to get and set that, but it's strange because you can use `name` everywhere except when you want to query on the db`.

Did we rule out using signals before to do this sort of thing? I think there's also a lifecycle hook you could compute and save the name in, too. Otherwise we have to remember to compute the name and save it every-time we instantiate an IngestRun.

Reply:
Yea we already do that here: https://github.com/kids-first/kf-api-study-creator/blob/8f8ee36693b9c2c2ff4110715f0c331b38d62605/creator/ingest_runs/signals.py#L22